aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -114,6 +115,33 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * If {@link org.apache.iceberg.aws.glue.GlueCatalog} should use external lock or not, default to false.
+   * If set to true, it will use DynamoDB to enforce locking during commits.
+   */
+  public static final String GLUE_CATALOG_LOCK_ENABLED = "gluecatalog.lock.enabled";
+  public static final boolean GLUE_CATALOG_LOCK_ENABLED_DEFAULT = false;
+
+  /**
+   * The DynamoDB table used for locking.
+   * One lock table is designed to be used for only one catalog.
+   * It is recommended to use a different table name for each Glue catalog.
+   * If the table does not exist, it will be created at runtime.
+   */
+  public static final String GLUE_CATALOG_LOCK_TABLE = "gluecatalog.lock.table";
+  public static final String GLUE_CATALOG_LOCK_TABLE_DEFAULT = "IcebergGlueCatalogLockTable";
```

Comment:
Can we base this on a prefix + catalog name instead of a constant? That way we don't do something unsafe by default.

Reply:
I considered this option, and decided to go with the way of setting the entire table name. This is because people can specify different catalog names and still point to the same Glue catalog, and in that case they want to use the same lock table. So setting the entire table name seems to be the most flexible option to satisfy all use cases.

Reply:
I'm just talking about the default value. If a user wants to set the table specifically, then I agree that it should be the lock.table property to override. But the default should be based on a prefix and the catalog name to avoid needless conflict.

Reply:
Combined with the discussion above, I think not providing a default name seems to be the right approach to go, so that user knows exactly what table they are using, and its cost associated. Let me update with that.